### PR TITLE
[DEV-1887] Add seo on all pages

### PR DESCRIPTION
**Pull request by tommaso1**
#### List of Changes

Add seo on all pages

#### Motivation and Context

The feature was missing

#### How Has This Been Tested?

Manually and running tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/lib/strapi/codecs/ProductCodec.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { MediaCodec } from './MediaCodec';
 import { NullToUndefinedCodec } from './NullToUndefinedCodec';
 import qs from 'qs';
 import { fetchFromStrapi } from '../fetchFromStrapi';
+import { SEOCodec } from './SeoCodec';
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/lib/strapi/codecs/ProductCodec.ts:6]:
```suggestion
```

File: apps/nextjs-website/src/lib/strapi/codecs/ProductCodec.ts [hunk 2]
```diff
@@ -23,6 +24,7 @@ export const ProductCodec = t.strict({
       description: t.union([NullToUndefinedCodec, t.string]),
       slug: t.string,
       logo: t.strict({ data: MediaCodec }),
+      seo: t.union([NullToUndefinedCodec, SEOCodec]),
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/lib/strapi/codecs/ProductCodec.ts:27]:
```suggestion
```

File: apps/nextjs-website/src/lib/strapi/__tests__/solutions.test.ts [hunk 3]
```diff
@@ -165,6 +167,7 @@ const makeStrapiResponseJson = () => ({
           {
             id: 1,
             attributes: {
+              seo: null,
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/lib/strapi/__tests__/solutions.test.ts:170]:
```suggestion
```

File: apps/nextjs-website/src/lib/strapi/__tests__/fixtures/solution.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { mediaVectorJson } from '@/lib/strapi/__tests__/fixtures/media';
 import { productsJson } from '@/lib/strapi/__tests__/fixtures/product';
+import { setOption } from 'monocle-ts/lib/Optional';
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/lib/strapi/__tests__/fixtures/solution.ts:3]:
```suggestion
```

File: apps/nextjs-website/src/lib/strapi/__tests__/fixtures/product.ts [hunk 5]
```diff
@@ -15,6 +15,7 @@ export const productJson = {
   data: {
     id: 8,
     attributes: {
+      seo: null,
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/lib/strapi/__tests__/fixtures/product.ts:18]:
```suggestion
```

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

File: .changeset/green-moons-jog.md
[the remark this answers is not in the document]
Comment (MarcoPonchia) [reply .changeset/green-moons-jog.md:1]: I also will add a new changeset dedicated to the changes of this pr

**Review by MarcoPonchia — CHANGES_REQUESTED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Review by marcobottaro — APPROVED** [review]

**Pull request merged**